### PR TITLE
updated documentation and some variables renaming

### LINKED ISSUE #142: Update KMS policy
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The resource types for kms has changed to keys. Need to update the policy statement so that OKE can use KMS.

### New or Affected Resource(s)

oci_identity_policy.oke-kms

### Potential Terraform Configuration

```hcl
# Copy-paste any Terraform configurations for how the requested feature may be used. 
```

### LINKED ISSUE #249: Support for api_private_key
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In my Terraform scripts I use the "private_key" attribute of the OCI Provider instead of the private_key_path because it is more convenient when used from CI/CD to express the private key as an environment variable rather than having to have some script in the CI/CD workflow write the private key to disk before invoking Terraform. I would like to be able to provide the raw private key to the OKE module as well.

### New or Affected Resource(s)

Module "oke"

### LINKED ISSUE #250: Support api_private_key_password
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In my terraform scripts, I use an API private key that includes a passphrase. The OCI provider has an attribute "private_key_password" which is used to set this. I don't seem to have something similar on the OKE module. I expected to find "api_private_key_password", but this resulted in an error.

### LINKED ISSUE #357: [v4.x] DOCS:
On v4.x branch, there's a warning:
```
│ Warning: Provider oci.home is undefined
│ 
│   on main.tf line 109, in module "oke-v4":
│  109:     oci.home = oci.home
│ 
│ Module module.oke-v4 does not declare a provider named oci.home.
│ If you wish to specify a provider configuration for the module, add an entry for oci.home in the required_providers block within the module.
```
obviously `versions.tf` misses alias definitions:
```
terraform {
  required_providers {
    oci = {
      source = "hashicorp/oci"
      configuration_aliases = [ oci.home ]
    }
  }
  required_version = ">= 1.0.0"
}
```

this has to be documented at level

### PR DESCRIPTION
**Pull request by hyder**
Closes #357 
Closes #249 
Closes #250 
Closes #142

### CODE REVIEW

File: docs/configuration.adoc [hunk 1]
```diff
@@ -128,7 +128,7 @@ The networking parameters concern the VCN and the subnets network configuration
 You can leave most of the default options. However, you may want to change the following 2 parameters:
 
 * vcn_dns_label: this is the internal dns domain for resources created
-* vcn_name: this is the name of the vcn that will be appended to the label prefix
+* vcn_name: this is the name of the vcn that will be prepended to the label prefix
```
Comment (karthicgit) [inline docs/configuration.adoc:131]: its appended to label_prefix

File: docs/topology.adoc [hunk 2]
```diff
@@ -84,16 +71,16 @@ The following resources are created:
 * 1 public regional bastion subnet
 * 1 private regional operator subnet
 
-The bastion subnet is regional i.e. in multi-AD regions, the subnet spans all Availability Domains. By default, the bastion subnet is assigned a CIDR of 10.0.0.0/30 giving a maximum possible of 1 IP address.
+The bastion subnet is regional i.e. in multi-AD regions, the subnet spans all Availability Domains. By default, the bastion subnet is assigned a CIDR of 10.0.0.0/29 giving a maximum possible of 1 IP address.
```
Comment (karthicgit) [inline docs/topology.adoc:74]: /29 so maximum possible is more than 1 IP address.

File: variables.tf [hunk 3]
```diff
@@ -522,48 +505,38 @@ variable "node_pool_os_version" {
...
-
-  validation {
-    condition     = contains(["public", "private"], var.worker_mode)
-    error_message = "Accepted values are public or private."
-  }
-}
-
 # oke load balancers
 
 variable "lb_type" {
```
Comment (karthicgit) [inline variables.tf:535]: the variable naming  is misleading . can it be changed to lbsubnet_type ?

File: modules/extensions/locals.tf [hunk 4]
```diff
@@ -25,7 +25,7 @@ locals {
   ## 1. bastion to be enabled and in a running state
   ## 2. operation to be enabled and instance_principal to be enabled
 
-  post_provisioning_ops = var.create_bastion_host == true && var.bastion_state == "RUNNING" && var.create_operator == true && var.operator_instance_principal == true ? true : false
+  post_provisioning_ops = var.create_bastion_host == true && var.bastion_state == "RUNNING" && var.create_operator == true && var.enable_operator_instance_principal == true ? true : false
```
Comment (karthicgit) [inline modules/extensions/locals.tf:28]: Do we need one more check for operator_state=RUNNING ?

File: variables.tf (hunk 3 above)
Comment (hyder) [reply variables.tf:535]: good point. I've changed for the preferred one as well.

File: modules/extensions/locals.tf (hunk 4 above)
Comment (hyder) [reply modules/extensions/locals.tf:28]: well spotted, thanks

**Review by karthicgit — APPROVED** [review]

**Pull request merged**